### DISCUSSION TITLE
Problem with hyphens in table names?

### DISCUSSION THREAD

**Discussion by Evgeny-Jem** [sqlalchemy/sqlalchemy#10190] [Usage Questions] [answered]
I was called to develop an external service to an existing project.
postgres database.
Described the models of the existing database using sglacodegen v2, then manually corrected.
When I tried to run the simplest function, I got an error:
```
async def get_user_details(user_id: str):
    user_uuid = uuid.UUID(user_id)
    async with get_session() as session:
        result = await session.execute(
            select(Users)
            .where(Users.id == user_uuid)
        )
        user_details = result.first()
        if user_details:
            return user_details[0]


if __name__ == '__main__':
    user_id = '00490294-63c0-47b9-bd35-3eddca4e54f7'
    user_detail = asyncio.run(get_user_details(user_id))
    print(f"ID: {user_detail.id}")
```
```
   File "D:\works_python\humart_app\venv\Lib\site-packages\sqlalchemy\orm\clsregistry.py", line 532, in __call__
     x = eval(self.arg, globals(), self._dict)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   File "<string>", line 1, in <module>
TypeError: unsupported operand type(s) for -: 'Table' and 'Table'
```
Full log is here: https://pastebin.com/PgGhvhNG

Then wrapped in quoted_name table names containing a hyphen wherever possible (table name, in column definitions, in relationships)

But nothing helps.
Where to look and how to solve the problem?

**Comment by gordthompson** [comment]
> Where to look and how to solve the problem?

You could start by providing an [MCVE](https://stackoverflow.com/help/minimal-reproducible-example).

Despite your post's title, you haven't mentioned anything about a table name, let alone one with a hyphen in it.

**Comment by Evgeny-Jem** [comment]
Closed until the discussion is complete and correct

**Comment by zzzeek** [comment, marked as answer]
hi - 

it looks like you are using sqlacodegen and it generated a class name that looks like `SomeTable-SomeTable`.

As you mentioned, you corrected the program manually for this case, so it looks like you missed a spot, likely it generated a relationship target like `relationship("SomeTable-SomeTable")` which got evaluated at runtime.  so you'd want to fix that.

also feel free to take this to [sqlacodegen](https://github.com/agronholm/sqlacodegen) since this seems a bug on their end, they should be escaping out hyphens.

**Reply by Evgeny-Jem** [reply]
I think the same. Now I'm looking for where I could overlook the error in order to fix it. It's just that there are more than 150 tables in the existing database, and there are a lot of relationships (relations). The search will take a long time. If I do not find an error, then I will issue a post more correctly and completely.

**Reply by zzzeek** [reply]
grep for the word "relationship()", that's where it will be

**Reply by zzzeek** [reply]
```bash
find myapp/ -name "*.py" -exec grep -H "relationship("  {} \;  | grep '-'
```

done
